src/lib/server/links/utils.test.ts

```diff
@@ -1,7 +1,215 @@
 import { test, expect } from 'vitest';
-import { parseLinkId } from './utils';
+import { parseLinkId, parseLink, findLinkId } from './utils';
 
-test('parses shortlinks', () => {
-  expect(parseLinkId('missing-link')).toBe(undefined);
-  expect(parseLinkId('GitHub')).toBe('github');
+const TEST_LINKS = {
+  example: 'https://example.com',
+  'example/abc': 'https://example.com/123/',
```

Comment:
Quality-of-life feature idea: Support known shortlink IDs as a prefix for shortlink destinations.

Given the shortlink `https://acmcsuf.com/green`, we expect a computed destination of `https://example.com/green` where destination value `/example/green` of link ID `green` computes to (`/example` → `https://example.com`) + `/green`.

```suggestion
  example: 'https://example.com',
  'example/abc': 'https://example.com/123/',
  'green': '/example/green',
```

Reply:
Added support for internal redirect resolution in recent commits.